Source/Charts/Renderers/BarChartRenderer.swift

```diff
@@ -896,4 +934,32 @@ open class BarChartRenderer: BarLineScatterCandleBubbleRenderer
 
         return element
     }
+
+    private func findTopRectInBar(barRects: [CGRect], firstIndexInBar: Int, lastIndexInBar: Int) -> CGRect {
+        var topRectInBar = barRects[firstIndexInBar]
+        if barRects[lastIndexInBar].origin.y < topRectInBar.origin.y {
+            topRectInBar = barRects[lastIndexInBar]
+        }
+        
+        var height: CGFloat = 0
+        for index in firstIndexInBar...lastIndexInBar {
+            height += barRects[index].height
+        }
+        
+        topRectInBar.size.height = height
+        
+        return topRectInBar
+    }
+
+    /// Creates path for bar in rect with rounded corners
+    internal func createBarPath(for rect: CGRect, roundedCorners: UIRectCorner) -> UIBezierPath {
+
+        let cornerRadius = rect.width / 2.0
+
+        let path = UIBezierPath(roundedRect: rect,
```

Comment:
@skarol I see a problem here when the rect is smaller than the cornerRadius, in this case you'll have  a smashed rounded rectangle that looks bad(I haven't tested this code, but I worked in this feature in my fork, and at the start I had used the UIBezierPath like you with the result that I have commented, some screenshots at the end, apologies for the quality). 
My proposal to solve this is to split it into two cases(Take into consideration that I haven't used the `open var roundedCorners: UIRectCorner = []`. I will always round all corners)
1. `rect.height >= rect.width` in this case we can perform the UIBezierPath as you do
2. `rect.height < rect.width` in this case we can draw a portion of a circumference, I leave the algorithm
```
let height = rect.size.height
let hypotenuse = width / 2.0
let radius = hypotenuse
let startAngle: CGFloat
let endAngle: CGFloat
if height < radius {
    let oLeg = radius - height
    startAngle = asin(oLeg/hypotenuse)
    endAngle = CGFloat(Double.pi) - startAngle
} else {
    let oLeg = height - radius
    startAngle = -asin(oLeg/hypotenuse)
    endAngle = CGFloat(Double.pi) - startAngle
}
let y = rect.origin.y - (width - rect.size.height)
let center = CGPoint(x: rect.origin.x + radius, y: y + radius)
let bezierPath = UIBezierPath(arcCenter: center, radius: radius, startAngle: startAngle, endAngle: endAngle, clockwise: true)
```
Before 
![before2](https://user-images.githubusercontent.com/912381/72202346-a3544300-345e-11ea-9dd9-32da6ffdfaff.png)
![before3](https://user-images.githubusercontent.com/912381/72202350-a818f700-345e-11ea-8151-194460c06642.png)

After
![share1](https://user-images.githubusercontent.com/912381/72202137-6424f280-345c-11ea-9d50-e9322323f876.png)
![share](https://user-images.githubusercontent.com/912381/72202139-6be49700-345c-11ea-9e55-64ff47e6ef6c.png)